tools/depends/target/Toolchain.cmake.in

```diff
@@ -32,23 +32,35 @@ elseif(OS STREQUAL osx)
   set(CORE_SYSTEM_NAME osx)
 elseif(OS STREQUAL ios)
   set(CMAKE_SYSTEM_NAME Darwin)
-  set(CORE_SYSTEM_NAME ios)
+  if(PLATFORM STREQUAL appletvos)
+    set(CORE_SYSTEM_NAME tvos)
+  else()
+    set(CORE_SYSTEM_NAME ios)
+  endif()
```

Comment:
I still think making tvos a complete separate platform is the wrong approach, but I would have to see the whole code to make a final decision.

Reply:
So I we should add all the core changes to this PR?

Reply:
Rechi, what do thing about keeping `CORE_SYSTEM_NAME` to `ios` and setting `CORE_PLATFORM_NAME` to `tvos` as described here https://github.com/SylvainCecchetto/xbmc/issues/42#issuecomment-485733563?

Reply:
At least having a branch which contains all the required changes and has no merge conflicts with master branch of this repo would enough. Also I don't see much use in splitting adding the tvOS code into multiple PRs as all of them are needed to get it working. Correctly splitting the commits is enough.

Yes I think using the `CORE_PLATFORM_NAME` approach which is already used on linux for the different windowings is the way to go. But as said I would like to see the whole code first.

If picking the `CORE_PLATFORM_NAME` way I would make three defines, `TARGET_DARWIN_IPHONE` (or maybe someone has a better name for this one), `TARGET_DARWIN_IOS` and `TARGET_DARWIN_TVOS`.

Reply:
Instead if `TARGET_DARWIN_IPHONE` how about `TARGET_DARWIN_EMBEDDED` or `TARGET_DARWIN_NONDESKTOP`

Reply:
TARGET_DARWIN_EMBEDDED sounds okay.

Reply:
Perfect, thanks